src/main/java/edu/harvard/iq/dataverse/FileDownloadServiceBean.java

```diff
@@ -242,9 +242,15 @@ private void redirectToBatchDownloadAPI(String multiFileString, Boolean download
     public void explore(GuestbookResponse guestbookResponse, FileMetadata fmd, ExternalTool externalTool) {
         ApiToken apiToken = null;
         User user = session.getUser();
-        if (user instanceof AuthenticatedUser) {
-            AuthenticatedUser authenticatedUser = (AuthenticatedUser) user;
-            apiToken = authService.findApiTokenByUser(authenticatedUser);
+        DatasetVersion version = fmd.getDatasetVersion();
+        if (version.isDraft() || (fmd.getDataFile().isRestricted())) {
+            if (user instanceof AuthenticatedUser) {
+                AuthenticatedUser authenticatedUser = (AuthenticatedUser) user;
+                apiToken = authService.findApiTokenByUser(authenticatedUser);
+                if ((apiToken == null) || (apiToken.getExpireTime().before(new Date()))) {
+                    apiToken = authService.generateApiTokenForUser(authenticatedUser);
```

Comment:
I know we added generateApiTokenForUser in pull request #5755 for the launching of external tools as part of #5671 (and added messaging to the users that this is happening) but are we intentionally adding it here in the download code as well? Soon all users who download anything will have API Tokens, right? Is this what we want? I don't see this in the notes for QA you just added (thanks for those!).

Reply:
That was only for config tools and this is only for explore tools (not downloads), and here it is only if those credentials are actually needed (i,e, not for published, unrestricted files). Nominally messaging could be added here too but that means new some code since the config too mechanism doesn't use the standard message panel. I can add that in, but some more discussion about the message might be useful (will a user understand an email about an api key when all they did was preview a file? How do we explain? Do we explain differently for creating or updating an api key? Do we explain it's in the URL of the previewer (regardless of whether we create a new one or just use one they had)?). It would also be good to discuss whether messaging changes if/when the api key is not in the URL.

Reply:
@qqmyers if it helps, here is the messaging I mentioned that was added in the older pull request:

- API Token will be generated. Please keep it secure as you would do with a password.
- API Token has been generated. Please keep it secure as you would do with a password.

(It should probably say "An API token". Oh well. 😄 .)

This is probably old news to you because now that I look at #5671 again I can see you participated quite a bit in the discussion! Thank you!